src/integration/java/io/pinecone/helpers/BuildUpsertRequest.java

```diff
@@ -44,6 +41,23 @@ public static UpsertRequest buildRequiredUpsertRequest(List<String> upsertIds, S
                 .build();
     }
 
+    public static UpsertRequest buildRequiredUpsertRequestByDimension(List<String> upsertIds, int dimension, String namespace) {
```

Comment:
I wanted to be able to generate vectors with a specific `dimension` rather than relying on the hard-coded `upsertData` in this class.

Reply:
Okay, I can cleanup and refactor when dealing with data plane tests